openedx/openedx-platform
# [BD-04] Convert Course, Section and Sequence XModules to XBlocks

Converts the Course, Section and Sequence XModules into XBlocks.

Part of [XModule to XBlock Conversion work](https://openedx.atlassian.net/wiki/spaces/AC/pages/1472790755/XModule+to+XBlock+Conversion).

**Testing Instructions**
* Before checking out this branch, make sure you have some test courses.
* Checkout this branch. In `make lms-shell` and `make studio-shell` run `pip install -e common/lib/xmodule` and `NO_PREREQ_INSTALL=1 paver update_assets` and then do `make lms-restart` and `make studio-restart`.
* Create courses. Add files, edit settings and confirm no errors are raised.
* Add, edit and delete Sections and Subsections and confirm no errors are raised.
* Export one of the existing courses. Import the archive in a new course and confirm that the course outline + settings of both the courses matches.

To test the `SplitTestBlock` changes:
* Add `split_test` to Advanced Components in Advanced Settings.
* Add a Content Experiment component to a unit. Click the Edit button and see that you can edit the fields and save them.

**MRO Analysis**
Descriptor MRO | Module MRO | Block MRO | Notes
------------ | ------------- | ---------------- | ------------
<class 'xmodule.seq_module.SequenceDescriptor'> | <class 'xmodule.seq_module.SequenceModule'> | <class 'xmodule.seq_module.SequenceBlock'> |
<class 'xmodule.seq_module.SequenceMixin'> | | <class 'xmodule.seq_module.SequenceMixin'> |
<class 'xmodule.seq_module.SequenceFields'> | <class 'xmodule.seq_module.SequenceFields'> | <class 'xmodule.seq_module.SequenceFields'> |
<class 'xmodule.seq_module.ProctoringFields'> | <class 'xmodule.seq_module.ProctoringFields'> | <class 'xmodule.seq_module.ProctoringFields'> |
<class 'xmodule.mako_module.MakoModuleDescriptor'> | | | Inherits from (MakoTemplateBlockBase, XModuleDescriptor) and has the get_html() method which has been implemented in the XBlock.
<class 'xmodule.mako_module.MakoTemplateBlockBase'> | | <class 'xmodule.mako_module.MakoTemplateBlockBase'> |
<class 'xmodule.xml_module.XmlDescriptor'> | | | Inherits from (XmlMixin, XModuleDescriptor) and doens't have any atttributes so we can remove it.
<class 'xmodule.xml_module.XmlMixin'> | | <class 'xmodule.xml_module.XmlMixin'> |
<class 'xmodule.xml_module.XmlParserMixin'> | | <class 'xmodule.xml_module.XmlParserMixin'> |
<class 'xmodule.x_module.XModuleDescriptor'> | | |
<class 'xmodule.x_module.XModuleDescriptorToXBlockMixin'> | | <class 'xmodule.x_module.XModuleDescriptorToXBlockMixin'> |
 | <class 'xmodule.x_module.XModule'> | |
 | <class 'xmodule.x_module.XModuleToXBlockMixin'> | <class 'xmodule.x_module.XModuleToXBlockMixin'> |
<class 'xmodule.x_module.HTMLSnippet'> | <class 'xmodule.x_module.HTMLSnippet'> | <class 'xmodule.x_module.HTMLSnippet'> |
<class 'xmodule.x_module.ResourceTemplates'> | | <class 'xmodule.x_module.ResourceTemplates'> |
<class 'xmodule.x_module.XModuleMixin'> | <class 'xmodule.x_module.XModuleMixin'> | <class 'xmodule.x_module.XModuleMixin'> |
<class 'xmodule.x_module.XModuleFields'> | <class 'xmodule.x_module.XModuleFields'> | <class 'xmodule.x_module.XModuleFields'> |
<class 'xblock.core.XBlock'> | <class 'xblock.core.XBlock'> | <class 'xblock.core.XBlock'> |
